src/features/amUI/users/useDebouncedFilteredRightHolders.ts

```diff
@@ -0,0 +1,76 @@
+import { getTotalNumOfPages, getArrayPage, debounce } from '@/resources/utils';
+import type { RightHolder } from '@/rtk/features/userInfo/userInfoApi';
+import { useGetRightHoldersQuery } from '@/rtk/features/userInfo/userInfoApi';
+import { useState, useMemo, useEffect } from 'react';
+
+const isSearchMatch = (searchString: string, rightHolder: RightHolder): boolean => {
+  const isNameMatch = rightHolder.name.toLowerCase().indexOf(searchString.toLowerCase()) > -1;
+  const isPersonIdMatch = rightHolder.personId === searchString;
+  const isOrgNrMatch = rightHolder.organizationNumber === searchString;
+  return isNameMatch || isPersonIdMatch || isOrgNrMatch;
+};
+
+export const useDebouncedFilteredRightHolders = (
+  searchString: string,
+  currentPage: number,
+  pageSize: number,
+) => {
+  const { data: rightHolders, isLoading: isFetching } = useGetRightHoldersQuery();
```

Comment:
Not important, but in rtkQuery, isFetching is actually a possible return which tells us if the queary is loading either to get the data the first time or on a refetch, while isLoading tells us if the query is loading the data for the very first time. I'm assuming you are intentionally using isLoading here, but could maybe renaming the variable to isFetching potentially cause confusion for later reading of the code?

Reply:
The loading-prop is not in use, so i remove t for now.